nmtwizard/preprocess/preprocess.py

```diff
@@ -159,119 +159,75 @@ def generate_preprocessed_data(config, corpus_dir, data_dir, result='preprocess'
                 num_samples += f.lines_kept
                 summary[f.base_name]["lines_filtered"] = f.lines_kept
 
+        if hasattr(sampler_consumer, "finalize"):
+            sampler_consumer.finalize(config)
+
         data_path = result_dir
 
     return data_path, train_dir, num_samples, summary, metadata
 
-def preprocess_file(config, input):
+class Processor(object):
 
-    preprocessor = FileProcessor(config)
+    def __init__(self, config, process_type="sampling", preprocess_exit_step=None):
```

Comment:
`process_type` should be an enum. See for example: https://github.com/OpenNMT/OpenNMT-tf/blob/master/opennmt/models/sequence_to_sequence.py#L22